rxjava/src/main/kotlin/com/mindera/skeletoid/rxjava/Observable.kt

```diff
@@ -39,7 +39,12 @@ fun <T : Any> Observable<T>.createUniqueConcurrentRequestCache(
         return requestMap[key] as Observable<T>
     }
 
-    val obs = this.doFinally {
+    val obs = this.doAfterNext{
+        requestMap.remove(key)
+    }
```

Comment:
Why this change?

Reply:
This was added to prevent concurrency issues, if just done at finally, if you are using heavy concurrency it would be removed after it should.

/cc @Jmvcaseiro